keps/sig-architecture/617-improve-kep-implementation/README.md

```diff
@@ -0,0 +1,221 @@
+consume KEPs.
+
+### Non-Goals
+
+- API Review process
+- Feature request triage
+- Developer guide
+
+## Proposal
+
+This proposal is somewhat omnibus.  Individual items are detailed below.
+
+### 1) Standardize an intra-KEP denotation for "not resolved"
+
+There is a perception that "merge and iterate" can be used to avoid dissent.
+If one topic is under debate, but other parts are acceptable, it should be
+possible to merge a KEP and iterate on the debatable point.  However, without
+reading the entire history of PRs on a KEP, exactly what is "under debate" is
+not clear.
+
+Requirements:
+  - not an HTML comment (so it is visible in the rendered markdown)
+  - unambiguous
+  - can wrap large bodies of text
+
+Parts of a KEP that are being merged without consensus shall be denoted as
+follows:
+
+```
+<<[UNRESOLVED optional text]>>
+Stuff that is under debate.
+<<[/UNRESOLVED]>>
+```
+
+The "optional text" can include a list of names or other context, but should be
+short.
+
+### 2) Change the names of KEPs
+
+The date-format in the name is not very human friendly and doesn't do anything
+for machines.  Instead, we propose to use the issue number of the enhancement
+issue as the KEP number.  KEP names would now be:
+`1234-short-human-friendly-string`.  This makes them easier to reference and
+remember: "KEP 1234" is easier than "KEP 20190903".  We will not retroactively
+change existing KEPs, so as to not break external links to them.
+
+### 3) Move KEPs from flat-files to a directory structure
+
+It's sometimes useful to have secondary documents alongside a KEP, for example
+image files.  To make this cleaner, new KEPs will be created in their own
+sub-directories of the SIG-specific directories. As per the above, a KEPs
+directory would be named as `1234-short-human-friendly-string`.  Inside that
+directory, the body of the KEP would be `README.md`.  Supporting documents and
+images would go in that directory.
+
+<<[UNRESOLVED]>>
+The simplest path would be to not retroactively move existing KEP files, to
+preserve inbound links.  On the other hand, we could make new dirs for existing
+KEPs, which would let us extract the metadata into `kep.yaml` which would be
+more consistent for tooling.  We could symlink the old names to the new
+`subdir/README.md` for inbound links.
+<<[/UNRESOLVED]>>
+
+### 4) Metadata split from KEP text
+
+Today, KEP metadata is a header on the markdown file.  This proposal moves that
+metadata to a new `kep.yaml` in each KEP directory (see above).  This file will
+contain the same information that was previously in the YAML front-matter.
+
+### 5) Remove some fields from metadata
+
+Almost no KEPs actually set `editor` at all.  This proposes to remove it from
+the template.
+
+The `last-updated` field can be derived from git history.
+
+The `superseded-by` field is never set and can be implied by the corresponding
+`replaces` field.
+
+### 6) Formalize one-enhancement == one-KEP
```

Comment:
Yes yes yes  🙏

Reply:
Yeah, I was under the impression that this was understood by the wider community, but we still get questions on it, so it makes sense to be explicit.

Reply:
it has not been understood, so this is a huge improvement.